errors out running on ubuntu 15.04 machine

```
$ docker run \
>   --volume=/:/rootfs:ro \
>   --volume=/var/run:/var/run:rw \
>   --volume=/sys:/sys:ro \
>   --volume=/var/lib/docker/:/var/lib/docker:ro \
>   --publish=8080:8080 \
>   --detach=true \
>   --name=cadvisor \
>   google/cadvisor:latest
ec7ae802836e362f50fab8897d85c1fad6e50029d8456e5018a17e87ea32af34
        
pczarkowski@fileserver:~$ docker logs cadvisor
panic: close of closed channel

goroutine 13 [running]:
github.com/godbus/dbus.(*Conn).Close(0x4c2081059e0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/godbus/dbus/conn.go:178 +0x62
github.com/godbus/dbus.(*Conn).inWorker(0x4c2081059e0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/godbus/dbus/conn.go:334 +0x10be
created by github.com/godbus/dbus.(*Conn).Auth
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/godbus/dbus/auth.go:118 +0xe84

goroutine 1 [runnable]:
fmt.(*ss).getRune(0x4c2081e8000, 0x4c200000066)
	/usr/lib/google-golang/src/fmt/scan.go:220
fmt.(*ss).token(0x4c2081e8000, 0x1, 0xb35408, 0x0, 0x0, 0x0)
	/usr/lib/google-golang/src/fmt/scan.go:472 +0x5d
fmt.(*ss).convertString(0x4c2081e8000, 0x73, 0x0, 0x0)
	/usr/lib/google-golang/src/fmt/scan.go:832 +0x11b
fmt.(*ss).scanOne(0x4c2081e8000, 0x73, 0x8280a0, 0x4c2081f3c50)
	/usr/lib/google-golang/src/fmt/scan.go:989 +0x1132
fmt.(*ss).doScanf(0x4c2081e8000, 0xa37eb0, 0x17, 0x4c2081ad758, 0x8, 0x8, 0x5, 0x0, 0x0)
	/usr/lib/google-golang/src/fmt/scan.go:1161 +0x373
fmt.Fscanf(0x7fc77dfe91e8, 0x4c2081bf890, 0xa37eb0, 0x17, 0x4c2081ad758, 0x8, 0x8, 0x7fc77dfd64b0, 0x0, 0x0)
	/usr/lib/google-golang/src/fmt/scan.go:145 +0xc0
fmt.Sscanf(0x4c2081d6500, 0xa0, 0xa37eb0, 0x17, 0x4c2081ad758, 0x8, 0x8, 0x8, 0x0, 0x0)
	/usr/lib/google-golang/src/fmt/scan.go:117 +0xcb
github.com/docker/docker/pkg/mount.parseInfoFile(0x7fc77dfe7fb0, 0x4c208034030, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/docker/pkg/mount/mountinfo_linux.go:61 +0x69d
github.com/docker/docker/pkg/mount.parseMountTable(0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/docker/pkg/mount/mountinfo_linux.go:39 +0x135
github.com/docker/docker/pkg/mount.GetMounts(0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/docker/pkg/mount/mount.go:8 +0x4c
github.com/docker/libcontainer/cgroups.FindCgroupMountpoint(0x9f72f0, 0x3, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/libcontainer/cgroups/utils.go:19 +0x5a
github.com/google/cadvisor/container/docker.func·001()
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/container/docker/factory.go:62 +0x50
sync.(*Once).Do(0x100a9c0, 0xb35778)
	/usr/lib/google-golang/src/sync/once.go:44 +0xd4
github.com/google/cadvisor/container/docker.UseSystemd(0x4c208189c50)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/container/docker/factory.go:70 +0x36
github.com/google/cadvisor/container/docker.Register(0x7fc77dfeed40, 0x4c2080c2180, 0x7fc77dfed3b0, 0x4c20814f2e0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/container/docker/factory.go:217 +0x887
github.com/google/cadvisor/manager.New(0x4c2080c6480, 0x7fc77dfe9168, 0x100a940, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/manager/manager.go:139 +0xa20
main.main()
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/cadvisor.go:69 +0x454

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1001b00)
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:839 +0x78
created by github.com/golang/glog.init·1
	/usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:406 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/lib/google-golang/src/runtime/asm_amd64.s:2232 +0x1

goroutine 11 [syscall]:
os/signal.loop()
	/usr/lib/google-golang/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/lib/google-golang/src/os/signal/signal_unix.go:27 +0x35
```

I can repro it as well; here's what I get:

```
I0428 15:16:37.709945 00001 storagedriver.go:92] No backend storage selected
I0428 15:16:37.710220 00001 storagedriver.go:94] Caching %d stats in memory60
I0428 15:16:37.710441 00001 manager.go:105] cAdvisor running in container: "/system.slice/docker-6489e42b05cf5e27eda698ee17fb3782948d30fbaa3cfa4503812df9c608687a.scope"
I0428 15:16:37.720195 00001 fs.go:87] Filesystem partitions: map[/dev/mapper/tungsten-home:{mountpoint:/rootfs/home major:252 minor:1} /dev/sda1:{mountpoint:/rootfs/boot major:8 minor:1} /dev/mapper/t
ungsten-root_x64:{mountpoint:/rootfs major:252 minor:4} /dev/mapper/tungsten-games:{mountpoint:/rootfs/games major:252 minor:3}]
E0428 15:16:37.731583 00001 machine.go:267] Failed to get system UUID: open /sys/class/dmi/id/product_uuid: no such file or directory
I0428 15:16:37.731756 00001 machine.go:223] Couldn't collect info from any of the files in "/etc/machine-id,/var/lib/dbus/machine-id"
I0428 15:16:37.731846 00001 manager.go:126] Machine: {NumCores:2 CpuFrequency:2933000 MemoryCapacity:8238407680 MachineID: SystemUUID: BootID:304919a0-724a-4b52-abd2-d20669a649fa Filesystems:[{Device:
/dev/sda1 Capacity:238787584} {Device:/dev/mapper/tungsten-root_x64 Capacity:27345022976} {Device:/dev/mapper/tungsten-games Capacity:21003628544} {Device:/dev/mapper/tungsten-home Capacity:8996602675
2}] DiskMap:map[8:16:{Name:sdb Major:8 Minor:16 Size:80026361856 Scheduler:deadline} 8:32:{Name:sdc Major:8 Minor:32 Size:150039945216 Scheduler:deadline} 252:0:{Name:dm-0 Major:252 Minor:0 Size:84137
73824 Scheduler:none} 252:2:{Name:dm-2 Major:252 Minor:2 Size:322122547200 Scheduler:none} 252:4:{Name:dm-4 Major:252 Minor:4 Size:27917287424 Scheduler:none} 8:0:{Name:sda Major:8 Minor:0 Size:500107
862016 Scheduler:deadline} 252:1:{Name:dm-1 Major:252 Minor:1 Size:91536490496 Scheduler:none} 252:3:{Name:dm-3 Major:252 Minor:3 Size:21474836480 Scheduler:none} 252:5:{Name:dm-5 Major:252 Minor:5 Si
ze:136365211648 Scheduler:none}] NetworkDevices:[{Name:eth0 MacAddress:00:23:ae:7a:96:20 Speed:100 Mtu:1500} {Name:lxcbr0 MacAddress:9e:93:dd:cc:f2:23 Speed:0 Mtu:1500}] Topology:[{Id:0 Memory:8238407
680 Cores:[{Id:0 Threads:[0] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1}]} {Id:1 Threads:[1] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:
1}]}] Caches:[]}]}
I0428 15:16:37.734790 00001 manager.go:133] Version: {KernelVersion:3.13.0-34-generic ContainerOsVersion:Buildroot 2014.02 DockerVersion:1.6.0 CadvisorVersion:0.11.0}
panic: close of closed channel

goroutine 18 [running]:
github.com/godbus/dbus.(*Conn).Close(0x4c2080b3560, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/godbus/dbus/conn.go:178 +0x62
github.com/godbus/dbus.(*Conn).inWorker(0x4c2080b3560)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/godbus/dbus/conn.go:334 +0x10be
created by github.com/godbus/dbus.(*Conn).Auth
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/godbus/dbus/auth.go:118 +0xe84

goroutine 1 [runnable]:
fmt.(*stringReader).Read(0x4c208029e00, 0x4c2080cbfd0, 0x1, 0x4, 0x1, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:90
io.ReadAtLeast(0x7fedca1c7610, 0x4c208029e00, 0x4c2080cbfd0, 0x1, 0x4, 0x1, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/io/io.go:298 +0xf1
io.ReadFull(0x7fedca1c7610, 0x4c208029e00, 0x4c2080cbfd0, 0x1, 0x4, 0x1, 0x0, 0x0)
        /usr/lib/google-golang/src/io/io.go:316 +0x6d
fmt.(*readRune).readByte(0x4c2080cbfb0, 0x40e665, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:341 +0x142
fmt.(*readRune).ReadRune(0x4c2080cbfb0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:357 +0x5d
fmt.(*ss).ReadRune(0x4c208120000, 0x4c200000000, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:201 +0xe5
fmt.(*ss).getRune(0x4c208120000, 0x4c200000065)
        /usr/lib/google-golang/src/fmt/scan.go:221 +0x28
fmt.(*ss).token(0x4c208120000, 0x1, 0xb1e6c8, 0x0, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:472 +0x5d
fmt.(*ss).convertString(0x4c208120000, 0x73, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:832 +0x11b
fmt.(*ss).scanOne(0x4c208120000, 0x73, 0x81d340, 0x4c208090430)
        /usr/lib/google-golang/src/fmt/scan.go:989 +0x1132
fmt.(*ss).doScanf(0x4c208120000, 0xa24370, 0x17, 0x4c2080dd758, 0x8, 0x8, 0x6, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:1161 +0x373
fmt.Fscanf(0x7fedca1c7610, 0x4c208029e00, 0xa24370, 0x17, 0x4c2080dd758, 0x8, 0x8, 0x7fedca1b5000, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:145 +0xc0
fmt.Sscanf(0x4c20802b720, 0x50, 0xa24370, 0x17, 0x4c2080dd758, 0x8, 0x8, 0x80, 0x0, 0x0)
        /usr/lib/google-golang/src/fmt/scan.go:117 +0xcb
github.com/docker/docker/pkg/mount.parseInfoFile(0x7fedca1c6fb0, 0x4c208124250, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/docker/pkg/mount/mountinfo_linux.go:61 +0x69d
github.com/docker/docker/pkg/mount.parseMountTable(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/docker/pkg/mount/mountinfo_linux.go:39 +0x135
github.com/docker/docker/pkg/mount.GetMounts(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/docker/pkg/mount/mount.go:8 +0x4c
github.com/docker/libcontainer/cgroups.FindCgroupMountpoint(0x9e4bb0, 0x3, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/docker/libcontainer/cgroups/utils.go:19 +0x5a
github.com/google/cadvisor/container/docker.func·001()
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/container/docker/factory.go:55 +0x50
sync.(*Once).Do(0xfec938, 0xb1ea38)
        /usr/lib/google-golang/src/sync/once.go:44 +0xd4
github.com/google/cadvisor/container/docker.UseSystemd(0x4c208028990)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/container/docker/factory.go:63 +0x36
github.com/google/cadvisor/container/docker.Register(0x7fedca1c9a70, 0x4c208148000, 0x7fedca1c80b8, 0x4c20813aac0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/container/docker/factory.go:211 +0x887
github.com/google/cadvisor/manager.New(0x4c20810a1c0, 0x7fedca1c7f48, 0xfec8b8, 0x0, 0x0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/manager/manager.go:138 +0xa20
main.main()
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/cadvisor.go:69 +0x454

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0xfe3ae0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:839 +0x78
created by github.com/golang/glog.init·1
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:406 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/lib/google-golang/src/runtime/asm_amd64.s:2232 +0x1

goroutine 10 [syscall]:
os/signal.loop()
        /usr/lib/google-golang/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/lib/google-golang/src/os/signal/signal_unix.go:27 +0x35
```

Not to sound like I'm stating the obvious here, but I'd guess it's a kernel setting; /sys/class/dmi/id/product_uuid doesn't exist on my system.

--- a/container/docker/factory.go
+++ b/container/docker/factory.go
@@ -24,7 +24,6 @@ import (
 	"sync"
 
 	"github.com/docker/libcontainer/cgroups"
-	"github.com/docker/libcontainer/cgroups/systemd"
 	"github.com/fsouza/go-dockerclient"
 	"github.com/golang/glog"
 	"github.com/google/cadvisor/container"
@@ -55,16 +54,19 @@ var check = sync.Once{}
 
 func UseSystemd() bool {
 	check.Do(func() {
-		// run and initialize useSystemd
-		useSystemd = systemd.UseSystemd()
-		if !useSystemd {
-			// Second attempt at checking for systemd, check for a "name=systemd" cgroup.
-			mnt, err := cgroups.FindCgroupMountpoint("cpu")
+		useSystemd = false
+
+		// Check for system.slice in systemd and cpu cgroup.
+		for _, cgroupType := range []string{"name=systemd", "cpu"} {
+			mnt, err := cgroups.FindCgroupMountpoint(cgroupType)
 			if err == nil {
 				// systemd presence does not mean systemd controls cgroups.
 				// If system.slice cgroup exists, then systemd is taking control.
 				// This breaks if user creates system.slice manually :)
-				useSystemd = utils.FileExists(mnt + "/system.slice")
+				if utils.FileExists(path.Join(mnt, "system.slice")) {
+					useSystemd = true
+					break
+				}
 			}
 		}
 	})

